src/textlint-rule-date-weekday-mismatch.js

```diff
@@ -65,7 +89,31 @@ function reporter(context, config = {}) {
     return {
         [Syntax.Str](node){
             const text = getSource(node);
-            const chronoDates = chrono.parse(text);
+            let chronoDates = chrono.parse(text);
+            // Add current year if missing and option is enabled
+            if (useCurrentYearIfMissing) {
+                const currentYear = (new Date()).getFullYear();
+                chronoDates.forEach(chronoDate => {
+                    // If year is not specified in the parsed result
+                    if (
+                        chronoDate.start &&
+                        chronoDate.start.knownValues.year === undefined
+                    ) {
+                        // Detect language for the date string
+                        const lang = detectLang(Object.keys(chronoDate.tags), preferLang);
+                        if (!lang) {
+                            return;
+                        }
+                        // Re-parse the text with the year added
+                        const newText = addYearToDateText(chronoDate.text, currentYear, lang);
+                        const reparsed = chrono.parse(newText, undefined, {forwardDate: true});
+                        // If reparsed successfully, update knownValues with year/month/day
+                        if (reparsed && reparsed[0] && reparsed[0].start && reparsed[0].start.knownValues) {
+                            Object.assign(chronoDate.start.knownValues, reparsed[0].start.knownValues);
+                        }
+                    }
+                });
+            }
```

Comment:
We avoid to use `let` and re-assign.
I think the scope of the let can be reduced by cutting it out to a function like createChronoDates.

```ts
const createChronoDates = (dateText:string, options: {
  preferLang
  useCurrentYearIfMissing: boolean
}) => { 
  let chronoDates = chrono.parse(text);
  if(!option.useCurrentYearIfMissing) {
    return chronoDates;
  }
  ... 
}

...
const chronoDates = createChronoDates(text, {
  useCurrentYearIfMissing
});
```